homeassistant/components/roborock/select.py

```diff
@@ -40,21 +41,23 @@ class RoborockSelectDescription(
     RoborockSelectDescription(
         key="water_box_mode",
         translation_key="mop_intensity",
-        options=RoborockMopIntensityCode.values(),
         api_command=RoborockCommand.SET_WATER_BOX_CUSTOM_MODE,
-        value_fn=lambda data: data.water_box_mode,
-        options_lambda=lambda data: [
-            k for k, v in RoborockMopIntensityCode.items() if v == data
+        value_fn=lambda data: data.water_box_mode.name,
+        options_lambda=lambda data: data.water_box_mode.keys()
+        if data.water_box_mode
+        else None,
+        parameter_lambda=lambda data: [
+            v for k, v in data[1].water_box_mode.items() if k == data[0]
```

Comment:
This appears to be doing a key lookup in the mode map. Is this equivalent to this?

```suggestion
            data[1].water_box_mode.get(data[0])
```

Reply:
good call - we don't have a direct get() on the roborockEnum class, but I can call as_dict() and then .get()